gbus/bus.go

```diff
@@ -648,19 +648,28 @@ func (b *DefaultBus) sendImpl(sctx context.Context, tx *sql.Tx, toService, reply
 		b.Log().WithError(err).Error("could not inject headers")
 	}
 
-	buffer, err := b.Serializer.Encode(message.Payload)
+	payloadBody, err := b.Serializer.Encode(message.Payload)
 	if err != nil {
 		b.Log().WithError(err).WithField("message", message).Error("failed to send message, encoding of message failed")
 		return err
 	}
 
+	if message.payloadBody != nil {
```

Comment:
If you plan on ignoring the `payloadBody` that you got out of the serializer, you should check for this before you run the encoder :)